near/kycdao-ntnft/src/lib.rs

```diff
@@ -124,8 +197,49 @@ impl KycdaoNTNFT {
         format!("{}/{}.json", base_uri, token_metadata.extra.expect("Missing token extra data"))
     }
 
+    /// Check the expiry of the token - None means it never expires
+    pub fn token_expiry(&self, token_id: TokenId) -> Option<u64> {
+        self.token_statuses.get(&token_id).unwrap_or_default().expiry
+    }
+
+    /// Check if the token is revoked
+    pub fn token_is_revoked(&self, token_id: TokenId) -> bool {
+        self.token_statuses.get(&token_id).unwrap_or_default().is_revoked
+    }
+
+    /// Check if an account has any valid tokens
+    pub fn has_valid_token(&self, address: AccountId) -> bool {
+        match self.tokens.tokens_per_owner.as_ref().expect("enumeration extension in use").get(&address) {
+            Some(token_ids) => {
+                for token_id in token_ids.iter() {
+                    let status = self.token_statuses.get(&token_id).unwrap_or_default();
+                    if status.is_valid() {
+                        return true;
+                    }
+                }
+                false
+            }
+            None => false,
+        }
+    }
+
     // TODO tokenVerificationURI???
 
+    /*****************
+    Admin
+    *****************/
+    pub fn set_revoke_token(&mut self, token_id: TokenId, is_revoked: bool) {
```

Comment:
Are these functions for updating revoke and expiry protected? I can't remember how NEAR implements it, do you need to be owner to write?

Reply:
good catch, I'll fix these soon and add regression tests for it